src/plugins/dashboard/public/dashboard_actions/clone_panel_action.tsx

```diff
@@ -89,17 +82,53 @@ export class ClonePanelAction implements Action<ClonePanelActionContext> {
       throw new PanelNotFoundError();
     }
 
-    dashboard.showPlaceholderUntil(
-      this.cloneEmbeddable(panelToClone, embeddable),
+    const clonedPanelState: PanelState<EmbeddableInput> = await (async () => {
+      const newTitle = await this.getCloneTitle(embeddable, embeddable.getTitle() || '');
+      const id = uuidv4();
+      if (isReferenceOrValueEmbeddable(embeddable)) {
+        return {
+          type: embeddable.type,
+          explicitInput: {
+            ...(await embeddable.getInputAsValueType()),
+            hidePanelTitles: panelToClone.explicitInput.hidePanelTitles,
+            title: newTitle,
+            id,
+          },
+        };
+      }
+      return {
+        type: embeddable.type,
+        explicitInput: {
+          ...panelToClone.explicitInput,
+          title: newTitle,
+          id,
+        },
+      };
+    })();
+    this.toastsService.addSuccess({
+      title: dashboardClonePanelActionStrings.getSuccessMessage(),
+      'data-test-subj': 'addObjectToContainerSuccess',
+    });
+
+    const { otherPanels, newPanel: placeholderPanelState } = createPanelState(
```

Comment:
**nit:** Should we rename this? Not sure that `placeholderPanelState` really fits anymore... Perhaps just `newPanelState`? Or `dashboardPanelState` (to communicate that `clonedPanelState` is of the generic type `PanelState`, while this new state is of type `DashboardPanelState`)?

Reply:
Good call, implemented this suggestion